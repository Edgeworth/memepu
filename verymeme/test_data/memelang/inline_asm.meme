// status: ok
// input: ''
// output: ''
// returns: 0
fn main() i8 {
  asm {
    lhu r3,5
  }
  asm{
    lhu r3,5
    lhu r3,5
  }
  asm
  {
    lhu r3,5
    lhu r3,5
    lhu r3,5
  }
  asm {}
  // a comment
  "a string";
  'a';
  ret 0;
}
