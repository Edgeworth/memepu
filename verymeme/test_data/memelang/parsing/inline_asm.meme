fn main() i8 {
  asm {
    lhu r3,5
  }
  asm{
    lhu r3,5
    lhu r3,5
  }
  asm
  {
    lhu r3,5
    lhu r3,5
    lhu r3,5
  }
  asm {}
  // a comment
  "a string";
  'a';
}
