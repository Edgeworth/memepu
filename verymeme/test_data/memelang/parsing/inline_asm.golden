File
 Fn
  FnSig
   Typename(main)
   Type(i8)
  StmtBlk
   Asm(
    lhu r3,5
  )
   Asm(
    lhu r3,5
    lhu r3,5
  )
   Asm(
    lhu r3,5
    lhu r3,5
    lhu r3,5
  )
   Asm()
   StrLit(a string)
   CharLit(0)
